### ISSUE TITLE
Can't pickle local object 'get_corrupt_function.<locals>.process'

### ISSUE THREAD

**Issue by v413945333** [TAMU-VITA/DeblurGANv2#44]
When I run train.py, this error happens.Does anyone have the same problem and how to solve it?
Pretrained models and datasets have been downloaded.
I have read some blogs, there may be the problems about multi thread.However I don't know where is wrong....seeking help

**Comment by lisishen** [comment]
Hello, I met this problem,too. Have you solve it, Thanks!

**Comment by Kai760** [comment]
I used
`get_dataloader = partial(DataLoader, batch_size=batch_size, shuffle=True, drop_last=True)`

instead of 
`get_dataloader = partial(DataLoader, batch_size=batch_size, num_workers=cpu_count(), shuffle=True, drop_last=True)`

**Comment by lisishen** [comment]
> I used
> get_dataloader = partial(DataLoader, batch_size=batch_size, shuffle=True, drop_last=True)
> instead of
> get_dataloader = partial(DataLoader, batch_size=batch_size, num_workers=cpu_count(), shuffle=True, drop_last=True)
> in train.py.

Oh! It worked!! Thanks a lot!!

**Comment by Sandbox3aster** [comment]
Hi, you can replace cpu_count() with 4 or 8 or the number of thread in you machine
`get_dataloader = partial(DataLoader, batch_size=batch_size, num_workers=4, shuffle=True, drop_last=True)`
